### PR TITLE
KOGITO-6399: DMN Expression Editor - undo decision table annotation column

### PR DESCRIPTION
**Pull request by danielzhe**
Ticket: https://issues.redhat.com/browse/KOGITO-6399

The undo/redo was not working when user changes the name of an annotation column because the `EditTextInline` inside the `TableHeader` was not notifying the GWT world about the user change.

### CODE REVIEW

**Review by jomarko — COMMENTED** [review]
I think we should add some sort of test @danielzhe

**Comment by yesamer** [comment]
@danielzhe Please add @fantonangeli anytime you changed something in React side, thank you!
@fantonangeli Can you please review it?

**Review by yesamer — APPROVED** [review]
LGTM Thank you @danielzhe

**Comment by fantonangeli** [comment]
Thank you for this PR, @danielzhe 
I agree with @jomarko 
Maybe we can re-enable and update the test in `DMNExpressionEditor.ts`:269 named "Test New Expresssion editor - undo"?

**Comment by danielzhe** [comment]
@fantonangeli We can't re-enable that test because of https://issues.redhat.com/browse/KOGITO-6402.

**Comment by jomarko** [comment]
@danielzhe then we need new test probably.

**Comment by yesamer** [comment]
@danielzhe Can you please resolve the conflict?

**Comment by danielzhe** [comment]
@fantonangeli @jomarko Changes applied.

**Comment by danielzhe** [comment]
I'm fixing the failing tests (that started to fail after I added new tests, which is very weird).

File: packages/boxed-expression-component/tests/components/DecisionTableExpression/DecisionTableExpression.test.tsx [hunk 1]
```diff
@@ -166,6 +166,7 @@ describe("DecisionTableExpression tests", () => {
       broadcastDecisionTableExpressionDefinition: (definition: DecisionTableProps) =>
         mockedBroadcastDefinition(definition),
       notifyUserAction: () => {},
+      selectObject(): void {},
```
Comment (danielzhe) [inline packages/boxed-expression-component/tests/components/DecisionTableExpression/DecisionTableExpression.test.tsx:169]: This is a try to fix the failing test. It was already failing before I added this.

**Comment by jomarko** [comment]
I wrote a message on google chat @danielzhe

**Comment by danielzhe** [comment]
All green now, @jomarko

**Review by fantonangeli — APPROVED** [review]
LGTM.
Thank you very much for this PR @danielzhe

**Review by jomarko — APPROVED** [review]
Thank you @danielzhe

**Review by yesamer — APPROVED** [review]

**Pull request merged**